### PR TITLE
Fix 300ms click delay for menu

### PR DESCRIPTION
**Pull request by robdodson**

### CODE REVIEW

**Comment by samccone** [comment]
wait... what I am so confused about this.. I thought on-tap was the right way to do things

**Comment by robdodson** [comment]
@samccone not in this case. We have handlers listening for `on-tap` to know when to close the drawer. So it hears the touch events, closes the drawer, then 300ms on iOS the `click` event fires, but now there's no longer an anchor in that position (because the drawer closed), so the link doesn't get clicked and we don't change pages. Discussed this with the Polymer team and the above fix is what they recommended.

**Comment by samccone** [comment]
so, having a blanket on-tap listener on the drawer regardless of the target seems like a bug in the drawer code to me.. like yes this is a fix, however it feels more like a hack around event propagation

**Comment by robdodson** [comment]
@samccone the `on-tap` listener was on the anchor clicks (not the drawer) but that's been switched to `on-click`

**Comment by samccone** [comment]
ok :)

I wonder if we might just want to handle the "click" ourselves in
https://github.com/PolymerElements/polymer-starter-kit/blob/7082366bc258ae748837d3adfe754e70955180f9/app/scripts/app.js#L73-L78

and then 

```js
e.preventDefault()
page.visit(e.currentTarget.....)
```

**Comment by robdodson** [comment]
@samccone PTAL. I moved the work into routing and also rewrote the routes so we're not duplicating the middleware usage.

File: app/scripts/app.js [hunk 1]
```diff
@@ -69,17 +69,13 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
...
-    }
-  };
-
   // Scroll page to top and expand header
   app.scrollPageToTop = function() {
     app.$.headerPanelMain.scrollToTop(true);
   };
 
+  app.closeDrawer = function() {
+    app.$.paperDrawerPanel.closeDrawer();
```
Comment (chuckh) [inline app/scripts/app.js]: @robdodson the `app.$.paperDrawerPanel.closeDrawer()` will always be called if narrow or not, but won't close the drawer when not narrow?

Comment (robdodson) [inline app/scripts/app.js]: `closeDrawer` doesn't seem to be doing anything fancy (https://github.com/PolymerElements/paper-drawer-panel/blob/master/paper-drawer-panel.html#L543) so I was thinking of just leaving out the narrow check to slim down the code. ¯\_(ツ)_/¯

Comment (chuckh) [inline app/scripts/app.js]: SGTM

Comment (addyosmani) [inline app/scripts/app.js]: SGTM2

**Comment by addyosmani** [comment]
LGTM but assigning to @samccone for a final call on the don't merge yet before we land this. Nice work, @robdodson :star:

**Comment by samccone** [comment]
:+1: good work all around :+1:

**Pull request merged**